src/c/soubory/otevirani_souboru.md

```diff
@@ -0,0 +1,157 @@
+# Otevírání souborů
+Abychom mohli s nějakým souborem začít pracovat, musíme ho nejprve v našem programu otevřít, aby
+byl vytvořen souborový deskriptor, do kterého pak můžeme zapisovat či z něho číst data. K tomu slouží
+funkce [`fopen`](https://devdocs.io/c/io/fopen), která má následující
+[signaturu](../funkce/funkce.md#syntaxe):
+```c
+FILE* fopen(const char* filename, const char* mode);
+```
+
+## Cesta k souboru
+Jako svůj první parametr funkce `fopen` očekává řetězec s cestou k souboru, který má být otevřen.
+Cestu můžete zadat dvěma způsoby:
+- **Absolutní cesta** (*absolute path*) je cesta, která začíná kořenovým adresářem souborového
+systému, například `/home/student/upr/soubor.txt`[^1]. Aby byla cesta absolutní, musí na Linuxu
+začínat dopředným lomítkem.
+- **Relativní cesta** (*relative path*) se vyhodnotí relativně k tzv. **pracovnímu adresáři**
+(*working directory*) běžícího programu. Pokud spustíte váš program z terminálu, tak se pracovní
+adresář implicitně nastaví na adresář, ze kterého jste program spustili. Pokud tedy například spustíte
+váš program ze složky `/home/student/upr` a funkci `fopen` předáte cestu `soubor.txt`, tak se funkce
+pokusí otevřít soubor na cestě `/home/student/upr/soubor.txt`.
+
+Při zadávání cesty můžete využít zkratky `.` a `..`, které jsou užitečné zejména u relativních cest:
+- Zkratka `.` se odkazuje na současný adresář, `./soubor.txt` je tedy to samé jako `soubor.txt`.
+- Zkratka `..` se odkazuje na rodičovský adresář, `../data/abc.txt` tedy říká:
+`Podívej se do rodičovské složky, tam nalezni adresář data a v něm soubor abc.txt`.
```

Comment:
Slozka a adresar. Chapu, ze cestinarsky je to lepsi, ale mozna bych slozku nahradil take adresarem. Ale je to na zvazeni.

Reply:
Jasně, to je chyba, snažil jsem se všude dávat adresář, ale tady mi to uniklo. Už jsem vytrénovaný od Standy, že složka je na Windowsech :D